readthedocs/projects/views/public.py

```diff
@@ -312,6 +314,24 @@ def get(self, request, project_slug, type_, version_slug):
             download=True,
         )
 
+    def _version_same_domain_url(self, request, type_, lang_slug, version_slug, subproject_slug=None):
+        final_project, lang_slug, version_slug, filename = _get_project_data_from_request(
+            request,
+            project_slug=None,
+            subproject_slug=subproject_slug,
+            lang_slug=lang_slug,
+            version_slug=version_slug,
+        )
+        return final_project.versions.get(slug=version_slug)
+
+    def _version_dashboard_url(self, request, project_slug, type_, version_slug):
+        version = get_object_or_404(
+            Version.objects.public(user=request.user),
+            project__slug=project_slug,
+            slug=version_slug,
+        )
+        return version
```

Comment:
I don't understand the logic changes at all in this file. What is `same_domain_url`? It isn't set anywhere here -- I assume it's being passed into the view, but feels weird and confusing. A docstring or comment above it would definitely help.

Reply:
This view is used to download a file using old-style URLs (download from the dashboard) and new-style URLs (download from the same domain as docs) --this is what `same_domain_url` is for.

Basically, the parameters received by the GET view are different (`project_slug` does not come in the new-style URLs, for example) and we need to take it from another place. Once we get the real `Version` to be served, everything is the same for both paths.

I'll add a comment.